__tests__/View-tests/home/maximiseScreen.test.tsx

```diff
@@ -65,6 +68,11 @@ describe("MaximizeScreen UI Tests", () => {
         postCaption: "Test Challenge",
         navigateGoBack: jest.fn(),
         userProfilePicture: "test_pp",
+        showGuestPopup: jest.fn(),
+        setShowGuestPopup: jest.fn(),
+        handleUserInteraction: jest.fn(() => {
+          mockToggleLike();
+        }),
```

Comment:
You could have centralize the user interaction with passing a parameter to this jest funtion as done in the real handleUserInteraction, to avoid spying on this viewmodel on specific cases